src/python/test/test_simplex_tree.py

```diff
@@ -249,3 +249,15 @@ def test_make_filtration_non_decreasing():
     assert st.filtration([3, 4, 5]) == 2.0
     assert st.filtration([3, 4]) == 2.0
     assert st.filtration([4, 5]) == 2.0
+
+def test_simplices_iterator():
+    st = SimplexTree()
+    
+    assert st.insert([0, 1, 2], filtration=4.0) == True
```

Comment:
Normally I am a bit wary of putting code with important side effects in assert: running with `python -O` disables the whole line. But this is just a test, and disabling the inserts just makes the loop below empty, so that should be fine.